### PR TITLE
Buildbot hintclass

### PR DESCRIPTION
**Pull request by nlsdvl**
Adds GF_ClassTypeHint enum to support clasifying filters in the documentation. 

https://github.com/gpac/wiki/pull/15#issuecomment-2422457046

### CODE REVIEW

**Comment by DenizUgur** [comment]
Were there any braking changes in this PR?

**Comment by jeanlf** [comment]
there should not for gpac, however wiki is now a bit messy -  we're working on it

File: src/Makefile [hunk 1]
```diff
@@ -220,9 +220,7 @@ LIBGPAC_FILTERS_FFMPEG=filters/ff_common.o filters/ff_avf.o filters/ff_dec.o fil
 LIBGPAC_FILTERS_LIBCAPTION=filters/dec_cc.o
 LIBGPAC_FILTERS_MPEGHDEC=filters/dec_mpeghdec.o
 
-ifeq ($(CONFIG_EMSCRIPTEN),yes)
```
Comment (rbouqueau) [inline src/Makefile:223]: Is that expected?

Comment (DenizUgur) [reply src/Makefile:223]: I guess not, there's no use of WebCodecs outside the Web right?

**Comment by DenizUgur** [comment]
> there should not for gpac, however wiki is now a bit messy -  we're working on it

Something broke in the WASM app when querying filters but I couldn't fully debug it.

File: src/Makefile (hunk 1 above)
Comment (jeanlf) [reply src/Makefile:223]: it is si that we generate the doc for all platforms

**Comment by jeanlf** [comment]
hmm strange there shouldn't be any change there, I'll look into it

**Comment by DenizUgur** [comment]
Let me debug my wasm build; it's different from the one in this repo.

**Pull request merged**